networkx/utils/backends.py

```diff
@@ -107,6 +92,65 @@
 Note that the ``backend_name`` is e.g. ``parallel``, the package installed
 is ``nx-parallel``, and we use ``nx_parallel`` while importing the package.
 
+.. _introspect:
+
+Introspection
+-------------
+Introspection techniques aim to demystify dispatching and conversion behaviors.
+They should help answer questions such as:
+
+- What happened (and why)?
+- What *will* happen (and why)?
+- Which backends implement a function?
+- What are the backend-specific arguments?
+- Where was time spent (including conversions)?
+- What is in the cache and how much memory is it using?
+
+We strive to make NetworkX backends easy to use, and we want them to "just work"
+for most users. Nevertheless, NetworkX dispatching should be transparent to allow
+for greater understanding, debug-ability, and customization. After all, NetworkX
+dispatching is extremely flexible and can support advanced workflows with multiple
+backends and fine-tuned configuration. It can support your needs as they evolve.
+
+Introspection capabilities are currently limited, but we are working to improve them.
+
+The primary way to see what the dispatch machinery is doing is by enabling logging.
+This can help you verify that the backend you specified is being used.
+You can enable NetworkX's backend logger to print to ``sys.stderr`` like this::
+
+    import logging
+    nxl = logging.getLogger("networkx")
+    nxl.addHandler(logging.StreamHandler())
+    nxl.setLevel(logging.DEBUG)
+
+And you can disable it by running this::
+
+    nxl.setLevel(logging.CRITICAL)
+
+Refer to `this <https://docs.python.org/3/library/logging.html>`_ to learn more about
+the logging facilities in Python.
+
+You can see which backends--that are currently installed--implement a function by
+looking at the ``.backends`` attribute, which is a set of backend names. For example::
+
+    >>> nx.betweenness_centrality.backends  # doctest: +SKIP
+    {'parallel'}
+
+The function docstring will also show which installed backends support it
+along with any backend-specific notes and keyword arguments::
+
+    >>> help(nx.betweenness_centrality)  # doctest: +SKIP
+    ...
+    Backends
+    --------
+    parallel : Parallel backend for NetworkX algorithms
+      The parallel computation is implemented by dividing the nodes into chunks
+      and computing betweenness centrality for each chunk concurrently.
+    ...
+
+The NetworkX documentation website also includes info about common backends in function references.
+For example, see `betweenness centrality <https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.centrality.betweenness_centrality.html>`_.
```

Comment:
How would you feel about doing this instead? This way it doesn't change to the online version if you're e.g. looking at a local doc build.

```suggestion
For example, see :func:`~networkx.algorithms.centrality.betweenness_centrality`.
```

Reply:
Thanks, let's give it a try! (is it obvious I'm still learning writing docs with sphinx :joy:?)

Reply:
worked :tada:

Reply:
Sweet! I'm figuring this stuff out too, for what it's worth. 😅

Reply:
Indeed it's always best to avoid explicit link targets when possible for no other reason than it helps future-proof the docs against broken links!